src/System.Buffers.Primitives/System/Buffers/Buffer.cs

```diff
@@ -26,11 +26,48 @@ internal Buffer(OwnedBuffer<T> owner, int index, int length)
             _length = length;
         }
 
-        internal Buffer(T[] array, int index, int length)
+        public Buffer(T[] array)
```

Comment:
I followed the [Span\<T\> implementation](https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.cs#L30). Does doing this have a performance cost? If not, should we also make the same change to Span\<T\>?

The longer one has this extra check:
```C#
if ((uint)start > (uint)array.Length || (uint)length > (uint)(array.Length - start))
    BufferPrimitivesThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
```